libs/utils/android/system.py

```diff
@@ -359,7 +359,7 @@ def gfxinfo_get(target, apk_name, out_file):
                     GET_FRAMESTATS_CMD.format(apk_name, out_file))
 
     @staticmethod
-    def monkey(target, apk_name, event_count=1):
+    def monkey(target, apk_name, event_count=1, pct_syskeys=0):
```

Comment:
There are many different turnings for different events:
   https://developer.android.com/studio/test/monkey.html
It would be nice to make this interface a bit more generic.

Perhaps we can  pass in a dictionary of percentages and map them on supported options.
Something like:

```python
def monkey(target, apk_name, event_count=1, percentages=None):
    PCT_PARAMS = ['touch', 'motion',  ...] # all supported events

    pcts = ''
    for evt, pct in percentages.iteritems():
        if evt not in PCT_PARAMS:
            continue
        if not 0 <= pct <= 100:
            raise ValueError("{} percentage out of [0..100] range")
        pcts += ' --pct-{} {}'.format(evt, value)
    #...
    target.execute('monkey {} -p {} {}'.format(pcts, apk_name, event_count))


# Then it should be possible to:

monkey(target, apk_name, event_count,
       percentages={'syskeys' : 0})


```